Rename `MModule` to `Module` and add constructor from `module` operation

jumerckx opened the pull request:
This enables creating modules with attributes such as:
```
module attributes {transform.with_named_sequence} { ... }
```

vchuravy commented:
Btw we could just name it `Module` instead of `MModule`.

vchuravy commented:
Could you also add a test?

Pangoraw commented on src/IR/IR.jl:
```diff
@@ -760,6 +760,7 @@ end
 
 MModule(loc::Location=Location()) =
     MModule(API.mlirModuleCreateEmpty(loc))
+MModule(op::Operation) = MModule(API.mlirModuleFromOperation(op))
```
I think we need this to prevent triggering both moduleDestroy and operationDestroy

```suggestion
MModule(op::Operation) = MModule(API.mlirModuleFromOperation(lose_ownership!(op)))
```

jumerckx commented:
I renamed `MModule` to `Module` and added tests, are they thorough enough?
Not sure how to get rid of: `WARNING: using IR.Module in module Main conflicts with an existing identifier.`

vchuravy commented:
Are we doing a `using LLVM`?

Pangoraw commented:
IIRC, I first called it `WModule` since otherwise it conflicted with `Base.Module` which is exported by `Base` and therefore `Module` was always ambiguous unless qualified.

We could do as LLVM and not export it:

```julia
julia> Base.isexported(LLVM, :Module)
false
```

jumerckx commented:
Thanks, not exporting does the trick.

Pangoraw commented:
I opened https://github.com/JuliaLang/julia/pull/53162 to allow doing:

```julia
using MLIR.IR: Module
```
